tests/test_yang.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
```

Comment:
Here's how I'm currently validating YANG vs. tree mapping in a different project:

```python
from pathlib import Path
import os
import subprocess
import pytest

TOPLEVEL = Path(__file__).parent.parent
YANG_FILES = [entries[0] + '/' + filename for entries in os.walk(TOPLEVEL / 'yang')
  for filename in entries[2] if filename.endswith('.yang')]

@pytest.mark.parametrize("yangfile", YANG_FILES)
def test_yang_tree(yangfile):
    res = subprocess.run(['pyang', '-f', 'tree', '-p', TOPLEVEL / 'yang', yangfile],
                         stdout=subprocess.PIPE)
    if res.returncode != 0:
        assert False, f'pyang failed: exit code {res.returncode}'
    treefile = Path(yangfile).with_suffix('.tree')
    tree = open(treefile, 'r').read()
    assert res.stdout.decode('utf-8') == tree, "YANG tree rendering differs"
```

If there's just one YANG file, then it's enough to only set `YANG_FILES = ['ietf-optical-impairment-topology.yang']` and get rid of the `/ 'yang'` when calling `subprocess.run`.

When running as `pytest -vv`, it prints out a nice diff on its own.

The only issue is that my favorite YANG tree rendering (CESNET's libyang's `yanglint`) shows default values as well, and `pyang` doesn't do that.